libraries/chain/apply_context.cpp

```diff
@@ -172,9 +176,11 @@ void apply_context::exec()
 {
    _notified.emplace_back( receiver, action_ordinal );
    exec_one();
+   trx_context.action_id.increment();
```

Comment:
Why is `action_id` being incremented?  Is this directly related to Deep Mind?  If not should it be pulled out into a separate PR?

Reply:
Yes it's related.. it's how actions are stitched together, the hierarchy is created.  It was there before the action ordinals were introduced.  Without action IDs, the stream of data cannot be reassembled.

Reply:
It's something we would prefer to keep as-is for now as it requires upstream changes and all our data models if removed.

We are open to switch to the `action_ordinal` in a future revision of the deep mind patch, it will require some planning on our side to refactor our data model and related code paths.